cmd/connectome2tck.cpp

```diff
@@ -39,7 +39,7 @@ using namespace MR::DWI;
 using namespace MR::DWI::Tractography;
 using namespace MR::DWI::Tractography::Connectome;
 
-const char *file_outputs[] = {"per_edge", "per_node", "single", NULL};
+const std::vector<std::string> file_outputs = {"per_edge", "per_node", "single"};
```

Comment:
Is `std::vector<std::string>` the best container to use here? These never change dynamically. I wonder if `std::array<std::string>` would be more appropriate? The code that resolves the user input against the available choices is regularly done by integer index, which precludes `std::set<std::string>`.

Reply:
Also, can they be `constexpr`?

Reply:
I probably should have mentioned this, but I did consider the use of `std::array<std::string,N>` in various places instead of `std::vector<std::string>` and ended up discarding the idea. Firstly because in our code, we often rely on `extern` declarations to allow a cpp file to define a certain variable. This implies that we would need to declare that variable for each `N` instantiated in the code, which would be cumbersome, to say the least. 

Secondly, classes like `MR::App::Argument` need to store lists of strings which means that the use `std::array<std::string>` would introduce the need for manual conversions to `std::vector<std::string>` (or alternatively require that those classes be templated the size of their lists, which is overkill IMO).

The use of `std::array` for this use case has very few benefits (e.g. performance is not even a concern) and would make the code less easy to maintain and read. 

For `constexpr` I believe we cannot do this until C++20 because `std::string` doesn't have a `constexpr` constructor in C++17. 

Additionally, an alternative to `std::vector<std::string>` would be `std::initializer_list<const char*>` to write something like:

```
// type is std::initializer_list<const char*>
constexpr auto file_outputs = {"per_edge", "per_node", "single"}; 
```
In C++20 one could even do:

```
using namespace std::string_literals;
// type is std::initializer_list<std::string>
constexpr auto file_outputs = {"per_edge"s, "per_node"s, "single"s}; 
```